src/eval/operation.rs

```diff
@@ -1115,6 +1122,85 @@ fn process_unary_operation(
                 ))
             }
         }
+        UnaryOp::Force(_) => {
+            /// `Seq` the `terms` iterator and then resume evaluating the `cont` continuation.
+            fn seq_terms<I>(terms: I, pos: TermPos, cont: RichTerm) -> RichTerm
+            where
+                I: Iterator<Item = RichTerm>,
+            {
+                terms
+                    .fold(cont, |acc, t| mk_app!(mk_term::op1(UnaryOp::Seq(), t), acc))
+                    .with_pos(pos)
+            }
+
+            match_sharedterm! {t,
+                with {
+                    Term::Record(map, attrs) if !map.is_empty() => {
+                        let mut shared_env = env.clone();
```

Comment:
Is there a reason to keep the old bindings of the environment? Unless I'm missing something, it's often fishy to closurize an environment inside itself. I think the reachable elements of the environment are only the thunks storing the elements of the record (or the array, below), and you allocate fresh thunks by closurization for the result. Thus, I think there is no code that can refer to the old elements of `env` inside `shared_end`?
```suggestion
                        let mut shared_env = Environment::new();
```

Reply:
It's not necessary at all. But it was in the earlier draft version. The same goes for `shared_env` in the `Array` match arm.